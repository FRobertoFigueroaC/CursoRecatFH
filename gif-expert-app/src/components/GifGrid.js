import React from 'react';
import {useFetchGifs} from "../hooks/useFetchGifs";
import {GifGridItem} from "./GifGridItem";
import PropTypes from 'prop-types';


export const GifGrid = ({ category}) => {

    const {data:images, loading} = useFetchGifs(category)


    return(
        <>
            <h3 className='animate__bounceIn'>{category}</h3>
            {loading && <p className='animate__fadeOutRight'>Loading</p>}
            <div className='card-grid'>
                <ul>
                    {
                        images.map(img => (
                            <GifGridItem key={img.id} {...img }/>
                        ))
                    }
                </ul>
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}