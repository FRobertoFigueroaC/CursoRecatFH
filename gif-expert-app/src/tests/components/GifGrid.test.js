import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el gif grid', () => {

    const category = 'Star Wars';
    
    test('debe mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading:true,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });


    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [
            {id: '1', url: 'www.something.com/img', title: 'Something'},
            {id: '2', url: 'www.something.com/img2', title: 'Something2'}
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading:false,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect( wrapper.find('p').exists()   ).toBe(false);

        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length)
        
    });
    
    
    
})
