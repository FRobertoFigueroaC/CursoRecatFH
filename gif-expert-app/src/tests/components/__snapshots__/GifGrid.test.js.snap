// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el gif grid debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__bounceIn"
  >
    Star Wars
  </h3>
  <p
    className="animate__fadeOutRight"
  >
    Loading
  </p>
  <div
    className="card-grid"
  >
    <ul />
  </div>
</Fragment>
`;
